applications/modal-bi-encoder/app.py

```diff
@@ -0,0 +1,23 @@
+from fastapi import FastAPI
+from src.main import (
+    encode,
+    EncodeInput,
+    EncodeOutput,
+    cosine_similarity,
+    CosineSimilarityInput,
+    CosineSimilarityOutput,
+)
+
+app = FastAPI(
+    title="Sentence Transformer: Bi-Encoder",
+)
+
+
+@app.post("/encode", response_model=EncodeOutput)
+def encode_request(input: EncodeInput):
+    return encode(input)
+
+
+@app.post("/cosine_similarity", response_model=CosineSimilarityOutput)
+def cosine_similarity_request(input: CosineSimilarityInput):
```

Comment:
this should look like the cohere api, check out my `sentence transformer` examples

Reply:
For this, there isn't a correlating API endpoint for `CosineSimilarity` in the Cohere (or OpenAI) API.

There is a [reranking function](https://docs.cohere.com/reference/rerank-1) in Cohere that is in the `sentence transformers` example, however since this current directory, `modal-bi-encoder` is only for bi-encoders, I don't believe it should implement any reranking function as that's  mainly for cross encoders. <https://www.sbert.net/examples/applications/retrieve_rerank/README.html> 

Thus, the Cohere-like reranking endpoint will instead be implemented in a future PR instead of this one, where the new `modal-cross-encoder` directory will be made that deals with fine-tuning a cross encoder.

Reply:
sounds good to me